### PR TITLE
[Bugfix] Gate DP-attention even-token padding to CP-enabled configs

### PR DESCRIPTION
**Pull request by Jiminator**
## Motivation

Fix a NaN crash (`Assertion 'NaN detected! draft_extend_for_prefill' failed`) in EAGLE/MTP speculative decoding with DP attention, introduced by #23269. It surfaces as a server crash on the `nightly-8-gpu-b200` test `TestEagleDPAttnServerLarge.test_a_gsm8k`.

## Modifications

In `ForwardBatch.prepare_mlp_sync_batch`, only pad `global_num_tokens` to a multiple of `attn_cp_size * 2` when context parallel is enabled (`attn_cp_size > 1`). #23269 changed this alignment from `attn_cp_size` to `attn_cp_size * 2`; with CP disabled (`attn_cp_size == 1`) that turned a no-op into even-length padding, injecting a zero-padded dummy token that the draft prefill-extend path mishandles and produces NaN on multi-seq prefill batches. With CP enabled the behavior is unchanged.

## Accuracy Tests

DeepSeek-V3-0324-FP4 EAGLE MTP, `--tp-size 4 --dp-size 4 --enable-dp-attention --attention-backend trtllm_mla`, GSM8K (200 ex): NaN crash (score ~0) before; **score ~0.97, accept_len ~3.0** after.

Verified end-to-end by dispatching the *Nightly Test (Nvidia)* workflow (`job_filter=nightly-test-specialized-8-gpu-b200`) against this branch — the previously-failing `TestEagleDPAttnServerLarge.test_a_gsm8k` now passes: :white_check_mark: https://github.com/sgl-project/sglang/actions/runs/26858581641

## Speed Tests and Profiling

N/A — removes unnecessary padding when CP is off; the CP path is unchanged.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #26841516075](https://github.com/sgl-project/sglang/actions/runs/26841516075)
Latest PR Test (Extra): :x: [Run #26851017754](https://github.com/sgl-project/sglang/actions/runs/26851017754)

### CODE REVIEW

File: python/sglang/srt/model_executor/forward_batch_info.py [hunk 1]
```diff
@@ -993,9 +993,14 @@ def prepare_mlp_sync_batch(self, model_runner: ModelRunner):
 
         # make sure that each rank has the same number of tokens to do collective communication and
         # we can divide the tokens into 2 * CP chunks for load balance.
+        # Only pad to 2*CP when CP is enabled; with CP off the extra even-length
+        # padding breaks EAGLE/MTP draft prefill (NaN draft logits, see #23269).
         attn_cp_size = get_attention_cp_size()
-        for i in range(sync_group_size):
-            global_num_tokens[i] = ceil_align(global_num_tokens[i], attn_cp_size * 2)
+        if attn_cp_size > 1:
```
Comment (kpham-sgl) [inline python/sglang/srt/model_executor/forward_batch_info.py]: Can you also guard this for zigzag mode only? Also left a small FIXME(kpham-sgl) note for me to take a look in the future

File: python/sglang/srt/layers/utils/cp_utils.py [hunk 2]
```diff
@@ -70,6 +68,23 @@ def is_prefill_cp_in_seq_split():
...
+    for zigzag (in-seq-split) CP, cp_size for other CP modes, and 1 when CP is
+    off so callers skip padding (extra padding breaks EAGLE/MTP draft prefill,
+    see #23269). Keep prepare_mlp_sync_batch and cal_padded_tokens consistent
+    through this helper.
+    """
+    from sglang.srt.layers.attention.dsa.utils import is_dsa_prefill_cp_in_seq_split
+
+    attn_cp_size = get_attention_cp_size()
+    if attn_cp_size <= 1:
+        return 1
```
Comment (kpham-sgl) [inline python/sglang/srt/layers/utils/cp_utils.py]: There shouldn't be any case where cp is disable (attn_cp_size == 1) and is_prefill_cp_in_seq_split() is True

File: python/sglang/srt/model_executor/forward_batch_info.py [hunk 3]
```diff
@@ -990,11 +993,15 @@ def prepare_mlp_sync_batch(self, model_runner: ModelRunner):
...
-            global_num_tokens[i] = ceil_align(global_num_tokens[i], attn_cp_size * 2)
+        # make sure that each rank has the same number of tokens to do collective communication.
+        # Zigzag (in-seq-split) CP pads to 2 * attn_cp_size for load balance; other CP modes
+        # pad to attn_cp_size; CP off pads nothing (extra padding breaks EAGLE/MTP draft
+        # prefill with NaN draft logits, see #23269).
+        # FIXME(kpham-sgl): revisit so draft prefill-extend tolerates padded dummy tokens.
+        cp_align_size = get_cp_padding_align_size()
+        if cp_align_size > 1:
+            for i in range(sync_group_size):
+                global_num_tokens[i] = ceil_align(global_num_tokens[i], cp_align_size)
```
Comment (kpham-sgl) [inline python/sglang/srt/model_executor/forward_batch_info.py]: Because ^ you don't need this if here
